### DISCUSSION TITLE
[Embedded App] Is re-rendering on first load expected behaviour?

### DISCUSSION THREAD

**Discussion by pnmcosta** [Shopify/polaris-react#9194] [Q&A] [answered]
When using the default node template https://github.com/Shopify/shopify-app-template-node/ if I set it up with WDYR as in here: https://github.com/welldone-software/why-did-you-render/issues/243#issuecomment-1181045461

I'm getting:

![image](https://github.com/Shopify/polaris/assets/550684/8ea24194-4589-4ce7-9690-1547794b9f97)

As soon as the app loads in the shop's admin, is this expected behaviour?

**Comment by pnmcosta** [comment]
Not sure if related or not, but any time my app loads via the test shop admin, there's a cancelled request to my apps url:

![image](https://github.com/Shopify/polaris/assets/550684/41315419-1bd8-47a6-805a-9952c92214be)

What's curious is the `ts` diff.

**Comment by pnmcosta** [comment]
This is more and more looking like a full refresh issue with app bridge itself, I'm going to report there too.

TLDR: any full page refresh (regardless of admin location) is querying all the apps installed:
![image](https://github.com/Shopify/polaris/assets/550684/156da86b-b810-44ab-a8fc-ae9f840a3ae3)

This is easy to replicate, make your network tab on browser developer tools to filter by a known domain host for any apps installed on your shopify store. 

If you open your admin home/products/discounts page and refresh I don't think you should see any request go through to the installed apps, but it is 🕵️

**Reply by TheSecurityDev** [reply, marked as answer]
I've seen this too. Basically it loads all the apps that are pinned on the navigation menu whenever the Shopify admin is loaded. It includes a query parameter `prefetch=1`, so I just handle requests with that parameter as 200 OK to improve performance.

**Reply by pnmcosta** [reply]
Thank you @TheSecurityDev, this seems to have helped resolve my original issue, I now prevent any react queries from occurring if `prefetch=1`. However does this really optimize the admin load times? The react apps is essentially just showing a spinner on this case, and I wonder if short circuiting it on the backend instead might be better?

**Reply by TheSecurityDev** [reply]
Yes, that's what I meant. In the backend before even loading the page just handle the response so you don't have to load the app in the background. I don't know why Shopify does this (perhaps to detect loading issues or attempt to optimize the loading by prefetching, but it doesn't help in most cases).
